text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/StyleSerializer.java

```diff
@@ -200,7 +204,7 @@ private boolean readBoolean(final JsonReader in) throws IOException {
   }
 
   private Object legacyHoverEventContents(final HoverEvent.Action<?> action, final Component rawValue) {
-    if (action == HoverEvent.Action.SHOW_TEXT) {
+    if (action == HoverEvent.Action.SHOW_TEXT || action == HoverEvent.Action.SHOW_ACHIEVEMENT) {
```

Comment:
I'm not entirely sure how the hover worked back then, but `rawValue` here is a `Component` and the function is only ever called when the action is `show_text` (it's only called when the action type is `Component`, that only corresponds to `show_text`), maybe I'm missing something but why is this needed if the `show_achievement` action of type `String`?